src/main/java/org/pesmypetcare/webservice/controller/petmanager/FreqTrainingRestController.java

```diff
@@ -0,0 +1,115 @@
+     */
+    @DeleteMapping("/{owner}/{petName}")
+    public void deleteAllFreqTrainings(@PathVariable String owner, @PathVariable String petName)
+        throws DatabaseAccessException {
+        freqTrainingService.deleteAllFreqTrainings(owner, petName);
+    }
+
+    /**
+     * Deletes the freqTraining with the specified pet and date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param petDate Date of the freqTraining instance
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @DeleteMapping("/{owner}/{petName}/{petDate}")
+    public void deleteFreqTrainingByDate(@PathVariable String owner, @PathVariable String petName,
+                                   @PathVariable String petDate) throws DatabaseAccessException {
+        freqTrainingService.deleteFreqTrainingByDate(owner, petName, petDate);
+    }
+
+    /**
+     * Get one freqTraining of the pet with the specified date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param petDate Date of the freqTraining instance
+     * @return The FreqTrainingEntity with the data of freqTraining
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}/{petDate}")
+    public FreqTrainingEntity getFreqTrainingByDate(@PathVariable String owner, @PathVariable String petName,
+                                        @PathVariable String petDate) throws DatabaseAccessException {
+        return freqTrainingService.getFreqTrainingByDate(owner, petName, petDate);
+    }
+
+    /**
+     * Gets all the specified freqTrainings of one pet.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @return The list with all of the freqTrainings
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}")
+    public List<Map<String, Object>> getAllFreqTraining(@PathVariable String owner, @PathVariable String petName)
+        throws DatabaseAccessException {
+        return freqTrainingService.getAllFreqTraining(owner, petName);
+    }
+
+    /**
+     * Gets the data from all the meals eaten by the pet between the initial and final date not including them.
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @param initialDate Initial Date
+     * @param finalDate Final Date
+     * @return The List containing all the freqTrainings in the specified time
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}/between/{initialDate}/{finalDate}")
+    public List<Map<String, Object>> getAllFreqTrainingsBetween(@PathVariable String owner, @PathVariable String petName,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 121).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5400494)